Software updates (#296)
AlexsLemonade/OpenPBTA-manuscript: content/06.methods.md

#### DNA WGS Alignment

We used `BWA-MEM` [@arxiv:1303.3997] to align paired-end DNA-seq reads to the version 38 patch release 12 of the _Homo sapiens_ genome reference, obtained as a FASTA file from UCSC (see **Key Resources Table**).
Next, we used the [Broad Institute's Best Practices](https://software.broadinstitute.org/gatk/best-practices/workflow?id=11165) to process Binary Alignment/Map files (BAMs) in preparation for variant discovery.
We marked duplicates using `SAMBLASTER` [@doi:10/f6kft3], and we merged and sorted BAMs using Sambamba [@doi:10/gfzsfw]

> We used the Broad's Genome Analysis Tool Kit [`GATK`](https://software.broadinstitute.org/gatk/), specifically the `BaseRecalibrator` submodule, to process BAMs. 

Author: ok. I think some of these software links were done in case where we didn't have pubs, but this has a pub, so I will add it and remove the link.

> Lastly, for normal/germline input, we ran the `GATK HaplotypeCaller` [@doi:10.1101/201178] submodule on the recalibrated BAM to generate a genomic variant call format (GVCF) file.

Author: @sjspielman once GATK versions are straightened out, do you want these added everywhere we use GATK in the manuscript or only in certain places?

This file is used as the basis for germline calling, described in the **SNV calling for B-allele Frequency (BAF) generation** section.

We obtained references from the [Broad Genome References on AWS](https://s3.console.aws.amazon.com/s3/buckets/broad-references/hg38/v0/) bucket with a general description of references at <https://s3.amazonaws.com/broad-references/broad-references-readme.html>.

#### Quality Control of Sequencing Data

To confirm sample matches and remove mis-matched samples from the dataset, we performed `NGSCheckMate` [@doi:10.1093/nar/gkx193] on matched tumor/normal CRAM files. 
Briefly, we processed CRAMs using `BCFtools` to filter and call 20k common single nucleotide polymorphisms (SNPs) using default parameters. 
We used the resulting VCFs to run `NGSCheckMate` following the `bcf_call.cwl` workflow found in the [D3b GitHub repository](https://github.com/d3b-center/OpenPBTA-workflows).
Per `NGSCheckMate` author recommendations, we used <= 0.61 as a correlation coefficient cutoff at sequencing depths > 10 to predict mis-matched samples.

> For RNA-Seq, we determined read strandedness by running the [`infer_experiment.py` script](http://rseqc.sourceforge.net/#infer-experiment-py) on the first 200k mapped reads.

Author: OK, it looks like this is actually a tool from a larger package called RNASeQC, so I will add that, but it also looks like [here](https://github.com/d3b-center/OpenPBTA-workflows/blob/1eded1969bdf34a14c9963acaa8eca3b9b2613cc/cwl/kfdrc_RNAseq_workflow.cwl#L704), we do not have a version. @migbro or @zhangb1, can you tell us what version of this tool is being used?

> For RNA-Seq, we determined read strandedness by running the [`infer_experiment.py` script](http://rseqc.sourceforge.net/#infer-experiment-py) on the first 200k mapped reads.

Author: Note @sjspielman I also created an issue [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1586) to add to our KR table supplement.

> For RNA-Seq, we determined read strandedness by running the [`infer_experiment.py` script](http://rseqc.sourceforge.net/#infer-experiment-py) on the first 200k mapped reads.

Reviewer: v2.3.4. The RNAseq workflow which is functionally equivalent has a bit better documentation here: https://github.com/kids-first/kf-rnaseq-workflow/tree/v2.4.2

We marked duplicates using `SAMBLASTER` [@doi:10/f6kft3], and we merged and sorted BAMs using `Sambamba` [@doi:10/gfzsfw]
We used the `BaseRecalibrator` submodule of the Broad's Genome Analysis Tool Kit `GATK` [@doi:10.1101/gr.107524.110] to process BAM files. 
Lastly, for normal/germline input, we used the `GATK` `HaplotypeCaller` [@doi:10.1101/201178] submodule on the recalibrated BAM to generate a genomic variant call format (GVCF) file.
This file is used as the basis for germline calling, described in the **SNV calling for B-allele Frequency (BAF) generation** section.

We obtained references from the [Broad Genome References on AWS](https://s3.console.aws.amazon.com/s3/buckets/broad-references/hg38/v0/) bucket with a general description of references at <https://s3.amazonaws.com/broad-references/broad-references-readme.html>.

#### Quality Control of Sequencing Data

To confirm sample matches and remove mis-matched samples from the dataset, we performed `NGSCheckMate` [@doi:10.1093/nar/gkx193] on matched tumor/normal CRAM files. 

> Briefly, we processed CRAMs using `BCFtools` to filter and call 20k common single nucleotide polymorphisms (SNPs) using default parameters. 

Author: not seeing one, only pubs using it to do comparisons or adapting the tool

We performed germline haplotype calls using the [GATK Joint Genotyping Workflow](https://software.broadinstitute.org/gatk/best-practices/workflow?id=11145) on individual GVCFs from the normal sample alignment workflow.
Using only SNPs, we applied the [GATK generic hard filter suggestions](https://gatkforums.broadinstitute.org/gatk/discussion/2806/howto-apply-hard-filters-to-a-call-set) to the VCF, with an additional requirement of 10 reads minimum depth per SNP.
We used the filtered VCF as input to `Control-FREEC` and `CNVkit` (below) to generate B-allele frequency (BAF) files. 

> Note that `GATK v4.0.12.0` was used for all of these steps except `VariantFiltration`, which used v3.8.0 because this tool was beta in `GATK v4.0.12.0` and known to be unreliable for this analysis.

Author: @migbro can you help out here as well - we have GATK version `v4.0.3.0` in the OpenPBTA-workflows repo for many steps, and have `v4.0.12.0` written in the manuscript, except for `v3.8.0` for `VariantFiltration`.

> Note that `GATK v4.0.12.0` was used for all of these steps except `VariantFiltration`, which used v3.8.0 because this tool was beta in `GATK v4.0.12.0` and known to be unreliable for this analysis.

Author: Also wanted to remind you @migbro there was a KF update to vardict which happened after we released the OpenPBTA and the data in OpenPBTA I believe use the older version.

> Note that `GATK v4.0.12.0` was used for all of these steps except `VariantFiltration`, which used v3.8.0 because this tool was beta in `GATK v4.0.12.0` and known to be unreliable for this analysis.

Reviewer: Yes, the results from VarDict 1.5 were used for CBTN data and any other data used at the time. Datasets following was 1.7. No plans to retroactively upgrade those 1.5 results.

> Note that `GATK v4.0.12.0` was used for all of these steps except `VariantFiltration`, which used v3.8.0 because this tool was beta in `GATK v4.0.12.0` and known to be unreliable for this analysis.

Author: Thanks. Any insight on GATK?

> Note that `GATK v4.0.12.0` was used for all of these steps except `VariantFiltration`, which used v3.8.0 because this tool was beta in `GATK v4.0.12.0` and known to be unreliable for this analysis.

Reviewer: The v3.8.0 for VariantFiltration still remains out of necessity as v4 did not provide that support.

#### Somatic Structural Variant Calling (WGS samples only)

We used `Manta` [@doi:10/gf3ggb] for structural variant (SV) calls, and we limited to regions used in `Strelka2`.
The hg38 reference for SV calling used was limited to canonical chromosome regions.
The somatic DNA workflows for SNV, indel, copy number, and SV calling can be found in the [OpenPBTA workflows Github repository](https://github.com/d3b-center/OpenPBTA-workflows/).
We used `AnnotSV` [@doi:10.1093/bioinformatics/bty304] to annotate `Manta` output.

> This associated workflow is available in the [workflows GitHub repository](https://github.com/d3b-center/OpenPBTA-workflows/).

Reviewer 2: Maybe we could save some space by changing this sentence, and delete line 187? For example, 

"All associated workflows described here are available in the [workflows GitHub repository](https://github.com/d3b-center/OpenPBTA-workflows/)."

Author: We do have unlimited space in methods, but I did update since it was redundant :)

For each multi-mapped gene symbol, we chose the Ensembl identifier corresponding to the maximum mean FPKM, using the assumption that the gene identifier with the highest expression best represented the expression of the gene.
After collapsing gene identifiers, 46,400 uniquely-expressed genes remained in the poly-A dataset, and 53,011 uniquely-expressed genes remained in the stranded dataset. 

##### Gene fusion detection

We set up `Arriba` [@doi:10.1101/gr.257246.119] and STAR-Fusion [@doi:10.1101/120295] fusion detection tools using CWL on CAVATICA.
For both of these tools, we used aligned BAM and chimeric SAM files from `STAR` as inputs and `GRCh38_gencode_v27` GTF for gene annotation.
We ran `STAR-Fusion` with default parameters and annotated all fusion calls with the `GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz` file from the `STAR-Fusion` release.
For `Arriba`, we used a blacklist file `blacklist_hg38_GRCh38_2018-11-04.tsv.gz` from the `Arriba` release to remove recurrent fusion artifacts and transcripts present in healthy tissue.
We provided `Arriba` with strandedness information for stranded samples, or we set it to auto-detection for poly-A samples.

> We used [`FusionAnnotator`](https://github.com/FusionAnnotator/FusionAnnotator) on `Arriba` fusion calls to harmonize annotations with those of `STAR-Fusion`.

Author: haha, didn't notice. I think it is linked to its own github since it is not published. Then, linked to our github for the repro workflow. Since this is forked to d3b, do you think leaving the latter is fine?

To determine the most focal units, we first considered the dominant status calls at the chromosome arm-level.
If the chromosome arm dominant status was callable but not clearly defined as a gain or loss, we instead included the cytoband-level status call.
Similarly, if a cytoband dominant status call was callable but not clearly defined as a gain or loss, we instead included gene-level status call.

> To obtain the gene-level data, we used the `mergeByOverlaps` [function](https://www.rdocumentation.org/packages/IRanges/versions/2.6.1/topics/findOverlaps-methods) from the `IRanges` package in R [@doi:10.1371/journal.pcbi.1003118] to find overlaps between the segments in the consensus CNV file and the exons in the GENCODE v27 annotation file (See **Key Resources Table**)	.

Reviewer 2: I would remove this link, and simply say we used the `IRanges` package in R. No need to reference the specific function.

If the copy number value was 0, we set the status to "deep deletion". 
For autosomes only, we set the status to "amplification" when the copy number value was greater than two times the ploidy value.

> We plotted genome-wide gains and losses in (**Figure {@fig:S3}B**) using the R package `ComplexHeatmap` [@doi:10.1093/bioinformatics/btw313].

Reviewer 2: What if we put these citations directly in the relevant captions? Would that save space? Like, "Panel B was made with `ComplexHeatmap` [@doi:10.1093/bioinformatics/btw313]," and similar.

Author: Methods doesn't count towards space, but I think figure captions may.. 

>* Resources should be under 45,000 characters (including spaces but not including the reference list, STAR Methods, or supplemental information). There may be up to 7 display items. Additional display items and materials may be published as supplemental information.

So it looks like legends may count. I have no idea what our count is now, so maybe we can assess once it is in the doc?

#### Chromothripsis Analysis (WGS samples only; `chromothripsis` analysis module)

Considering only chromosomes 1-22 and X, we identified candidate chromothripsis regions in the set of independent tumor WGS samples with ShatterSeek [@doi:10.1038/s41588-019-0576-7], using Manta SV calls that passed all filters and consensus CNV calls. 

> We modified the consensus CNV data to fit `ShatterSeek` input requirements as follows: CNV-neutral or excluded regions (both annotated as `NA` in the consensus data) were filled in with the respective sample’s ploidy value from `Control-FREEC`, and consecutive segments with the same copy number value were merged. 

Reviewer 2: ```suggestion
We modified the consensus CNV data to fit `ShatterSeek` input requirements as follows to set CNV-neutral or excluded regions as the respective sample’s ploidy value from `Control-FREEC`, and we then merged consecutive segments with the same copy number value. 
```

We performed molecular subtyping on tumors in the OpenPBTA to the extent possible.
The `molecular_subtype` field in `pbta-histologies.tsv` contains molecular subtypes for tumor types selected from `pathology_diagnosis` and `pathology_free_text_diagnosis` fields as described below, following World Health Organization 2016 classification criteria [@doi:10.1007/s00401-016-1545-1].

> Medulloblastoma (MB) subtypes SHH, WNT, Group 3, and Group 4 were predicted using the consensus of two RNA expression classifiers: `medullo-classifier-package` [@doi:10.1371/journal.pcbi.1008263] and `MM2S` [@doi:10.1186/s13029-016-0053-y] on the RSEM FPKM data (`molecular-subtyping-MB` analysis module).

Reviewer 2: It looks like the package itself is named `MedulloClassifier` but the repo is `medullo-classifier-package`, so we should probably use the former here.
